implementation.js

```diff
@@ -1,8 +1,43 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var global = require('system.global')();
+var isCallable = require('is-callable');
+var iterator = typeof Symbol === 'function' ? Symbol.iterator : '@@iterator';
```

Comment:
this is not sufficient. we can only use the string form when that's what `for..of` uses. In addition, there are some browsers that have NEITHER of these ways to determine iterability, and only piping through `for..of` can detect that.